plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiPageSourceProvider.java

```diff
@@ -192,7 +195,7 @@ public ConnectorPageSource createPageSource(
                 dataSourceStats,
                 options.withSmallFileThreshold(getParquetSmallFileThreshold(session))
                         .withVectorizedDecodingEnabled(isParquetVectorizedDecodingEnabled(session)),
-                timeZone);
+                timeZone, dynamicFilter);
```

Comment:
Could this lead to incorrect results for a MOR table that has both base and log files in the same file slice?  The dynamic filter is only applied on the parquet base file; the snapshot query merges base and log files for the actual records, on which the dynamic filter should be applied.

Do we know what types of dynamic filters are pushed down from Trino? E.g., filter on record keys is not affected since record keys are not changed by updates in the log files.

Reply:
Good catch, this would indeed be a problem. Let me verify this with a test.

Reply:
Sg.

Reply:
For file slices containing both the base and log files, there could still be a way to filter based on the dynamic filters and column stats with relaxed data skipping while maintaining the correctness (see relevant logic in Spark: https://github.com/apache/hudi/blob/42c1228928794b3c689849d495fe5d21abe1a45f/hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieFileIndex.scala#L263).  We can address this in a separate PR.